neqo-crypto/src/aead.rs

```diff
@@ -158,6 +191,41 @@ impl RealAead {
         }?;
         Ok(&output[0..(l.try_into()?)])
     }
+
+    /// Decrypt a ciphertext in place.
+    ///
+    /// Note that NSS insists upon having extra space available for decryption, so
+    /// the buffer for `output` should be the same length as `input`, even though
+    /// the final result will be shorter.
+    ///
+    /// # Errors
+    ///
+    /// If the input isn't authenticated or any input is too large for NSS.
+    pub fn decrypt_in_place(
+        &self,
+        count: u64,
+        aad: Range<usize>,
+        input: Range<usize>,
+        data: &mut [u8],
```

Comment:
As before, I would prefer a slice for the AAD.  However, in this case, you can insulate the caller from knowing about `self.expansion()`, so returning the `usize` is maybe OK (or you could return a slice).